src/RemoveUndef.cpp

```diff
@@ -106,52 +104,54 @@ class RemoveUndef : public IRMutator {
             } else {
                 predicate = cond;
             }
-            expr = t;
+            return t;
         } else if (cond.same_as(op->condition) &&
             t.same_as(op->true_value) &&
             f.same_as(op->false_value)) {
-            expr = op;
+            return op;
         } else {
-            expr = Select::make(cond, t, f);
+            return Select::make(cond, t, f);
         }
     }
 
-    void visit(const Load *op) {
+    Expr visit(const Load *op) override {
         Expr pred = mutate(op->predicate);
-        if (!pred.defined()) return;
+        if (!pred.defined()) return op;
```

Comment:
I think these should all be return Expr(), not return op. In this mutator, if you get an undefined Expr() back, you're supposed to pass it upwards as undefined.

Reply:
To mimic existing behavior, it should return whatever the default value of this->expr is... which (looks at code) is undefined. So, yeah.